### PR TITLE
Allow DSPO test suite be running with RHOAI

### PR DESCRIPTION
**Pull request by diegolovison**
## The issue resolved by this Pull Request:
Break down the solution for #[RHOAIENG-10212](https://issues.redhat.com//browse/RHOAIENG-10212) into smaller PRs to facilitate review.

## Description of your changes:
The goal of `test.sh --rhoai` is to be to run the DSPO test suite with a previous RHOAI installed.

## Testing instructions
* Have an OpenShift cluster with RHOAI installed and in the terminal do the following
* oc login ...
* export GIT_WORKSPACE=.....
* test.sh --rhoai

The output must be

```
--- PASS: TestIntegrationTestSuite (350.21s)
    --- PASS: TestIntegrationTestSuite/TestAPIServerDeployment (1.89s)
        --- PASS: TestIntegrationTestSuite/TestAPIServerDeployment/Should_successfully_fetch_pipelines (0.47s)
        --- PASS: TestIntegrationTestSuite/TestAPIServerDeployment/Should_successfully_upload_a_pipeline (0.91s)
        --- PASS: TestIntegrationTestSuite/TestAPIServerDeployment/Should_successfully_upload_a_pipeline_with_custom_pip_server (0.50s)
    --- PASS: TestIntegrationTestSuite/TestDSPADeployment (1.05s)
        --- PASS: TestIntegrationTestSuite/TestDSPADeployment/with_default_MariaDB_and_Minio (1.05s)
            --- PASS: TestIntegrationTestSuite/TestDSPADeployment/with_default_MariaDB_and_Minio/should_have_6_pods (0.45s)
            --- PASS: TestIntegrationTestSuite/TestDSPADeployment/with_default_MariaDB_and_Minio/should_have_a_ready_ds-pipeline-dspa-ext_deployment (0.30s)
            --- PASS: TestIntegrationTestSuite/TestDSPADeployment/with_default_MariaDB_and_Minio/should_have_a_ready_ds-pipeline-persistenceagent-dspa-ext_deployment (0.15s)
            --- PASS: TestIntegrationTestSuite/TestDSPADeployment/with_default_MariaDB_and_Minio/should_have_a_ready_ds-pipeline-scheduledworkflow-dspa-ext_deployment (0.15s)
    --- PASS: TestIntegrationTestSuite/TestFetchExperiments (0.15s)
        --- PASS: TestIntegrationTestSuite/TestFetchExperiments/Should_successfully_fetch_experiments (0.15s)
    --- PASS: TestIntegrationTestSuite/TestPipelineSuccessfulRun (327.14s)
        --- PASS: TestIntegrationTestSuite/TestPipelineSuccessfulRun/Should_create_a_Pipeline_Run (248.24s)
        --- PASS: TestIntegrationTestSuite/TestPipelineSuccessfulRun/Should_create_a_Pipeline_Run_using_custom_pip_server (78.90s)
PASS
ok  	github.com/opendatahub-io/data-science-pipelines-operator/tests	350.219s

```
## Checklist
- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/scripts/tests/tests.sh [hunk 1]
```diff
@@ -165,34 +166,54 @@ run_tests() {
...
 
-clean_up() {
+undeploy_kind_resources() {
   echo "---------------------------------"
   echo "Clean up"
   echo "---------------------------------"
   ( cd $GIT_WORKSPACE && make undeploy-kind )
 }
 
+remove_namespace_created_for_rhoai() {
```
Comment (HumairAK) [inline .github/scripts/tests/tests.sh:186]: can we get a comment here on why this is needed? and a log output as well? 

I'm guessing this is to cleanup old artifacts from a previous run?

File: .github/scripts/tests/tests.sh [hunk 2]
```diff
@@ -202,7 +223,6 @@ run_kind_tests() {
   apply_pip_server_configmap
   run_tests
   run_tests_dspa_external_connections
-  clean_up
 }
 # ------------------------------------
```
Comment (HumairAK) [inline .github/scripts/tests/tests.sh:227]: nit: these comments here look messy, can you get rid of these `# ----` lines? https://github.com/opendatahub-io/data-science-pipelines-operator/blob/f3871a3dcf01b151e4a849ff062f05a1c9a0ef77/.github/scripts/tests/tests.sh#L227-L243

Comment (diegolovison) [reply .github/scripts/tests/tests.sh:227]: For me, after spending some time looking at the source, it seems easy to understand the separation of responsibilities.

File: .github/scripts/tests/tests.sh (hunk 1 above)
Comment (diegolovison) [reply .github/scripts/tests/tests.sh:186]: > I'm guessing this is to cleanup old artifacts from a previous run?

There is a special task in the tracking document about the cleanup. We can discuss about that when we start working on the cleanup. Example:
* Should we do on the begin or in the end of the function?
* What should be default?
* Should we set a parameter or not?

For now, it is on the beginning to ensure that we won't have any flaky test.

File: .github/scripts/tests/tests.sh [hunk 3]
```diff
@@ -30,13 +26,19 @@ DSPA_PATH="${GIT_WORKSPACE}/tests/resources/dspa-lite.yaml"
...
 RESOURCES_DIR_PYPI="${GIT_WORKSPACE}/.github/resources/pypiserver/base"
 # ------------------------------------
 
 # ------------------------------------
 # Functions
+get_dspo_image() {
+  if [ "$REGISTRY_ADDRESS" = "" ]; then
+      echo "REGISTRY_ADDRESS variable not defined." && exit 1
+  fi
+  echo "${REGISTRY_ADDRESS}/data-science-pipelines-operator"
```
Comment (HumairAK) [inline .github/scripts/tests/tests.sh:39]: nit: can you provide more description in the echo statement

Comment (diegolovison) [reply .github/scripts/tests/tests.sh:39]: The echo is how you return a value in a bash script

Comment (HumairAK) [reply .github/scripts/tests/tests.sh:39]: I think you misunderstood. I'm saying to echo more than just the image, for example: 
```bash
echo "Using ${REGISTRY_ADDRESS}/data-science-pipelines-operator for DSPO image"  
```

File: .github/scripts/tests/tests.sh (hunk 2 above)
Comment (HumairAK) [reply .github/scripts/tests/tests.sh:227]: disagree, I think it's fairly common to have opts handling at the bottom or top in a bash script, I find the comments contribute to clutter, we can simply just leave the "# Run" comment, or expand on it like "# Script Entry point", or wrap it in a main()

File: .github/scripts/tests/tests.sh (hunk 1 above)
Comment (HumairAK) [reply .github/scripts/tests/tests.sh:186]: my suggestion is to include more description in the code, and add a log message like an echo statement explaining the step or some feedback for the user

File: .github/scripts/tests/tests.sh (hunk 2 above)
Comment (diegolovison) [reply .github/scripts/tests/tests.sh:227]: OK. going to not fight by that if you have a strong opition.

File: .github/scripts/tests/tests.sh (hunk 1 above)
Comment (diegolovison) [reply .github/scripts/tests/tests.sh:185]: Done

File: .github/scripts/tests/tests.sh (hunk 3 above)
Comment (diegolovison) [reply .github/scripts/tests/tests.sh]: Done

File: .github/scripts/tests/tests.sh (hunk 2 above)
Comment (diegolovison) [reply .github/scripts/tests/tests.sh]: Done

File: .github/scripts/tests/tests.sh [hunk 4]
```diff
@@ -30,13 +25,18 @@ DSPA_PATH="${GIT_WORKSPACE}/tests/resources/dspa-lite.yaml"
...
 CONFIG_DIR="${GIT_WORKSPACE}/config"
 RESOURCES_DIR_CRD="${GIT_WORKSPACE}/.github/resources"
-DSPO_IMAGE="${REGISTRY_ADDRESS}/data-science-pipelines-operator"
 OPENDATAHUB_NAMESPACE="opendatahub"
 RESOURCES_DIR_PYPI="${GIT_WORKSPACE}/.github/resources/pypiserver/base"
-# ------------------------------------
 
-# ------------------------------------
-# Functions
+# Do not add other echo in the bash function, otherwise it will return it as well
```
Comment (HumairAK) [inline .github/scripts/tests/tests.sh:31]: remove this

**Pull request merged**